lib/cc/engine/bundler_audit/analyzer.rb

```diff
@@ -15,10 +16,14 @@ def initialize(directory:, engine_config_path: DEFAULT_CONFIG_PATH, stdout: STDO
 
         def run
           raise(GemfileLockNotFound, "No Gemfile.lock found.") unless gemfile_lock_exists?
-          return unless gemfile_lock_in_include_paths?
+          unless gemfile_lock_in_include_paths?
+            stderr.puts("Not auditing #{gemfile_lock_relative_path} because it is not in include_paths")
+            return
+          end
 
           Dir.mktmpdir do |dir|
             FileUtils.cp(gemfile_lock_path, File.join(dir, GEMFILE_LOCK))
+            FileUtils.cp(gemfile_path, File.join(dir, GEMFILE))
```

Comment:
Yep, it's needed. Previous to https://github.com/codeclimate/codeclimate-bundler-audit/commit/6855d082f9659633b2209d679e909c41628e1be4, we were running the engine while the working directory contained both `Gemfile` and `Gemfile.lock`, and everything worked. After that change, we were running the engine while the working directory contained only `Gemfile.lock`, which leads to an error in bundler:

```
/usr/lib/ruby/gems/2.2.0/gems/bundler-1.10.5/lib/bundler.rb:198:in `rescue in root': Could not locate Gemfile or .bundle/ directory (Bundler::GemfileNotFound)
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-1.10.5/lib/bundler.rb:194:in `root'
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-1.10.5/lib/bundler.rb:210:in `app_cache'
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-1.10.5/lib/bundler/source/rubygems.rb:23:in `initialize'
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-1.10.5/lib/bundler/lockfile_parser.rb:34:in `new'
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-1.10.5/lib/bundler/lockfile_parser.rb:34:in `initialize'
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-audit-0.5.0/lib/bundler/audit/scanner.rb:42:in `new'
        from /usr/lib/ruby/gems/2.2.0/gems/bundler-audit-0.5.0/lib/bundler/audit/scanner.rb:42:in `initialize'
        from /usr/src/app/lib/cc/engine/bundler_audit/analyzer.rb:29:in `new'
        from /usr/src/app/lib/cc/engine/bundler_audit/analyzer.rb:29:in `block (2 levels) in run'
        from /usr/src/app/lib/cc/engine/bundler_audit/analyzer.rb:28:in `chdir'
        from /usr/src/app/lib/cc/engine/bundler_audit/analyzer.rb:28:in `block in run'
        from /usr/lib/ruby/2.2.0/tmpdir.rb:88:in `mktmpdir'
        from /usr/src/app/lib/cc/engine/bundler_audit/analyzer.rb:24:in `run'
        from /usr/src/app/bin/bundler-audit:7:in `<main>'
```

We weren't hitting the error because of the early return.